src/common/net.spec.ts

```diff
@@ -9,43 +9,88 @@ chai.use(chaiAsPromised)
 
 describe("Net", () => {
   const port = 4567
-  const host = "0.0.0.0"
+  const defaultHost = "0.0.0.0"
   const specialPort = process.platform.match("win") ? -1 : 80
 
   describe("#isPortAvailable", () => {
     context("when the port is not allowed to be bound", () => {
       it("returns a rejected promise", () => {
-        return expect(isPortAvailable(specialPort, host)).to.eventually.be
-          .rejected
+        return expect(isPortAvailable(specialPort, defaultHost)).to.eventually
+          .be.rejected
       })
     })
 
     context("when the port is available", () => {
       it("returns a fulfilled promise", () => {
-        return expect(isPortAvailable(port, host)).to.eventually.be.fulfilled
+        return expect(isPortAvailable(port, defaultHost)).to.eventually.be
+          .fulfilled
       })
     })
 
     context("when the port is unavailable", () => {
-      it("returns a rejected promise", () => {
-        createServer(port).then((_: { close(): any }) => {
-          return expect(isPortAvailable(port, host)).to.eventually.be.rejected
-        })
-      })
+      let closeFn = (cb: any) => cb()
+
+      it("returns a rejected promise", () =>
+        createServer(port).then((server: { close(): any }) => {
+          closeFn = server.close.bind(server)
+          return expect(isPortAvailable(port, defaultHost)).to.eventually.be
+            .rejected
+        }))
+
+      // close the servers used in this test as to not conflict with other tests
+      afterEach(done => closeFn(done))
+    })
+
+    // this test uses the test port on all ipv4 and ipv6 hosts via the respective broadcast
+    // addresses (0.0.0.0 and ::)
+    context("when the no local hosts are available", () => {
+      let closeIpV4ServerFn = (cb: any) => cb()
+      let closeIpV6ServerFn = (cb: any) => cb()
+
+      it("return a rejected promise", () =>
+        createServer(port, defaultHost).then((ipv4Server: { close(): any }) => {
+          closeIpV4ServerFn = ipv4Server.close.bind(ipv4Server)
+
+          return createServer(port, "::").then(
+            (ipv6Server: { close(): any }) => {
+              closeIpV6ServerFn = ipv6Server.close.bind(ipv6Server)
+              return expect(isPortAvailable(port, "127.0.0.1")).to.eventually.be
```

Comment:
I think this is meant to be `"::"`, not `"127.0.0.1"` ?

Reply:
For this test `127.0.0.1` is intended.

---
At line 51, the ipv4 hosts are listened on using the broadcast address (`0.0.0.0`) and line 54 is doing the same for ipv6.

This ensures that all of the checks in `isPortAvailable` will fail.

---
This line (line 57) is just testing with a host that we know is already taken, although I could have used any ipv4 or ipv6 address. However purposely didn't use `localhost`, just to avoid any shenanigans if the person running the test has done something unexpected with their hosts.

Reply:
For me (and in CI) this test doesn't pass. Does it pass for you?

Reply:
It passes locally for me (on node v12)
I will look into it further - I suspect there is something different between the node versions.

I should be able to explore over the weekend - will keep you posted

Reply:
Ooft. The flag `ipv6Only: true` in server.listen({}) only got introduced in node version 11.
My tests and implementation are leaning on that flag being used.

---
Apologies! Node version compatibility totally slipped my mind, I'll get that sorted ASAP

Reply:
Much appreciated, thanks!